### PR TITLE
Team: Improve a way to open team project and backup file creation

### PR DESCRIPTION
**Pull request by miurahr**
Improve a way to open team project

## Pull request type

Please check the type of change your PR introduces:

- [x] Bug fix
- [ ] Feature
- [ ] Documentation
- [ ] Build and release changes
- [ ] Other (describe below)

## Which ticket is resolved?

- Link: https://sourceforge.net/p/omegat/bugs/1110/
- Title:  Team project pile upback up files of omegat.project

## What does this PR change?

- Update `ProjectUICommands#projectOpen` method
    - Simplify detection of condition whether override local or not.
    - add missing message entries in `Bundle.properties`
   - Detect remote omegat.project file is identical with local one and if identical it does not make backup

### CODE REVIEW

File: src/org/omegat/util/FileUtil.java [hunk 1]
```diff
@@ -112,15 +101,12 @@ public int compare(File f1, File f2) {
 
     /**
      * Create file backup with datetime suffix.
-     *
-     * @return Backup file
      */
-    public static File backupFile(File original) throws IOException {
+    public static void backupFile(File original) throws IOException {
```
Comment (miurahr) [inline src/org/omegat/util/FileUtil.java:105]: Should not change public method for API stability.
We should use return value in caller class.

File: src/org/omegat/util/FileUtil.java [hunk 2]
```diff
@@ -78,31 +78,20 @@ private FileUtil() {
...
-                            return 1;
-                        } else {
-                            return 0;
-                        }
-                    }
-                });
+                Arrays.sort(bakFiles, (f1, f2) -> Long.compare(f2.lastModified(), f1.lastModified()));
                 for (int i = maxBackups; i < bakFiles.length; i++) {
-                    bakFiles[i].delete();
+                    boolean ignored = bakFiles[i].delete();
```
Comment (miurahr) [inline src/org/omegat/util/FileUtil.java:94]: Can we ignore a failure of removal of backup files?

**Comment by miurahr** [comment]
@brandelune here is a core part of team project, please assigns reviewer.

**Comment by brandelune** [comment]
Who is qualified to review this code?

**Comment by miurahr** [comment]
> Who is qualified to review this code?

From git blame, ISHIKAWA @zephyrus00jp , @amake and me touched this part.

File: src/org/omegat/gui/main/ProjectUICommands.java [hunk 3]
```diff
@@ -550,48 +540,21 @@ c. remote mapping, no local mapping(s)
...
-                            if (needToSavePropertiesFinal) {
-                                Log.logWarningRB("REMOTE_PROJECT_CONTENT_WAS_MODIFIED_TO_ADD_MAPPING_SO_NOT_USED");
-                                // Do not save the content of the remote file because we missed mapping.
-                                // So in our new scheme of backup and updating, we don't need to save the file
-                                // after this code block any more.
-                                needToSaveProperties = false;
-                                Files.deleteIfExists(rewriteOnSuccessFinal.toPath()); // remove "omegat.project.NEW"
-                            }
+                    // loading modified new project property
+                    final boolean succeeded = ProjectFactory.loadProject(props, true);
```
Comment (miurahr) [inline src/org/omegat/gui/main/ProjectUICommands.java:544]: Previous commit bcee3c8 tell us call `ProjectFactory.loadProject` should protect with `Core.executeExclusively`

https://github.com/omegat-org/omegat/blob/bcee3c8cf6ad9cef5789237ee60c353f10e994cc/src/org/omegat/gui/main/ProjectUICommands.java#L526-L542

Comment (brandelune) [reply src/org/omegat/gui/main/ProjectUICommands.java:544]: Thank you for tagging them.

**Comment by miurahr** [comment]
This passed integration test that duration is 20min.

**Comment by miurahr** [comment]
I'd like to merge and ask users feedback on Nightly version.

**Comment by brandelune** [comment]
Ok, let's do it.

**Comment by miurahr** [comment]
merged into `master`

**Pull request closed without merging**